### PR TITLE
ci: Move to GCC 5.4 (Xenial default) for prebuilts

### PR DESCRIPTION
**Pull request by mattklein123**
This is part 1 of a change to switch us over to a GCC 5.4 so that
we get a thread safe std::string implementation. Will follow up
with another commit with CI and doc changes.

### CODE REVIEW

**Comment by danielhochman** [comment]
+1

File: ci/build_container/build_container.sh [hunk 1]
```diff
@@ -5,13 +5,7 @@ set -e
...
-  clang-format-3.6 bc libtool automake zip time
-apt-get install -y golang
-# For debugging.
-apt-get install -y gdb strace
-add-apt-repository -y ppa:ubuntu-toolchain-r/test
-apt-get update
-apt-get install -y g++-4.9
+  clang-format-3.6 bc libtool automake zip time golang g++ gdb strace
 # clang head (currently 5.0)
 wget -O - http://apt.llvm.org/llvm-snapshot.gpg.key | apt-key add -
```
Comment (danielhochman) [inline ci/build_container/build_container.sh]: we could save some time by doing all the repo adding and coalescing the apt-get update

File: ci/build_container/build_container.sh [hunk 2]
```diff
@@ -32,10 +26,6 @@ pip install virtualenv
 export GOPATH=/usr/lib/go
 go get github.com/bazelbuild/buildifier/buildifier
 
-# GCC 4.9 for everything.
-export CC=gcc-4.9
-export CXX=g++-4.9
-
```
Comment (htuch) [inline ci/build_container/build_container.sh]: Is it worth invoking gcc here for a version check to ensure we have the version we think we do?

File: ci/build_container/build_container.sh (hunk 1 above)
Comment (mattklein123) [inline ci/build_container/build_container.sh]: So I tried this, but wget isn't installed by default. Sigh.

File: ci/build_container/build_container.sh (hunk 2 above)
Comment (mattklein123) [inline ci/build_container/build_container.sh]: Yeah might as well. I will add an echo that we can look at in the job. I actually think default GCC in Xenial might be 5.3.

Comment (htuch) [inline ci/build_container/build_container.sh]: It's probably still good to see `CC=gcc` etc. just to ensure that we don't accidentally get defaulted into clang one day by Bazel magic.

**Comment by mattklein123** [comment]
@htuch updated

File: ci/build_container/build_container.sh [hunk 3]
```diff
@@ -32,9 +26,10 @@ pip install virtualenv
 export GOPATH=/usr/lib/go
 go get github.com/bazelbuild/buildifier/buildifier
 
-# GCC 4.9 for everything.
-export CC=gcc-4.9
-export CXX=g++-4.9
+# GCC for everything.
+export CC=gcc
+export CXX=g++
+g++ --version
```
Comment (htuch) [inline ci/build_container/build_container.sh]: Could also do a `[ $(g++ --version) == "g++ ... 5.3.blah" ] || (echo "Unexpected compiler version"; exit 1)

**Comment by htuch** [comment]
Nit: maybe use CXX when invoking the compiler..

File: ci/build_container/build_container.sh (hunk 3 above)
Comment (mattklein123) [inline ci/build_container/build_container.sh]: This requires a bunch of sed stuff, I can try to figure this out.

Comment (htuch) [inline ci/build_container/build_container.sh]: Just compare the exact string - this is only used in the container where that will be stable.

**Comment by mattklein123** [comment]
@htuch updated

**Pull request merged**